WordPress/gutenberg
# Adds content locking to navigation block

## What?


The navigation block, when locked, cannot restrict editing of contents (links and 
submenus). This PR fixes this.

## Why?


Because it should work at least on the same feature level as a reusable block, which 
has this ability to restrict editing of contents.

## How?


Copies the same way in which the reusable blocks have the feature, thus also adding 
a new block api method `isNavigationBlock`.

## Testing Instructions






1. Add a post
2. Insert a navigation block
3. Insert some links
4. From the more menu (three dots) choose lock
- Does "Restrict editing" appear?
6. Check Restrict editing
7. Click Apply

- Does the setting apply? You should not be able to modify the navigation block's 
contents.

## Screenshots or screencast 



https://user-images.githubusercontent.com/107534/194833645-ab27fee8-b1d4-41d3-a993-74b9ec3f6330.mp4




## To do

- [x] Make block locking disable block reordering via list view as well
- [ ] Refactor the way we figure out entity based blocks available for locking
